src/main/java/io/fabric8/maven/docker/access/KeyStoreUtil.java

```diff
@@ -41,28 +43,40 @@ public static KeyStore createDockerKeyStore(String certPath) throws IOException,
         return keyStore;
     }
 
-    public static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
-        PEMKeyPair keyPair = loadPEM(keyPath);
-        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyPair.getPrivateKeyInfo().getEncoded());
-        return KeyFactory.getInstance("RSA").generatePrivate(keySpec);
+    static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
+        try (Reader reader = new FileReader(keyPath);
+            PEMParser parser = new PEMParser(reader)) {
+            Object readObject = parser.readObject();
+            if (readObject instanceof PEMKeyPair) {
+                PEMKeyPair keyPair = (PEMKeyPair) readObject;
+                return generatePrivateKey(keyPair.getPrivateKeyInfo());
+            } else if (readObject instanceof PrivateKeyInfo) {
+                return generatePrivateKey((PrivateKeyInfo) readObject);
+            }
+        }
+        return null;
     }
 
-    private static PEMKeyPair loadPEM(String keyPath) throws IOException {
-        PEMParser parser = new PEMParser(new BufferedReader(new FileReader(keyPath)));
-        return (PEMKeyPair) parser.readObject();
+    private static PrivateKey generatePrivateKey(PrivateKeyInfo keyInfo) throws IOException, NoSuchAlgorithmException,
+            InvalidKeySpecException {
+        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyInfo.getEncoded());
+        return KeyFactory.getInstance("RSA").generatePrivate(keySpec);
     }
 
-    private static void addCA(KeyStore keyStore, String caPath) throws KeyStoreException, FileNotFoundException, CertificateException {
+    private static void addCA(KeyStore keyStore, String caPath) throws IOException, KeyStoreException,
+            CertificateException {
         for (Certificate cert : loadCertificates(caPath)) {
             X509Certificate crt = (X509Certificate) cert;
             String alias = crt.getSubjectX500Principal().getName();
             keyStore.setCertificateEntry(alias, crt);
         }
     }
 
-    private static Certificate[] loadCertificates(String certPath) throws FileNotFoundException, CertificateException {
-        InputStream is = new FileInputStream(certPath);
-        Collection<? extends Certificate> certs = CertificateFactory.getInstance("X509").generateCertificates(is);
-        return new ArrayList<>(certs).toArray(new Certificate[certs.size()]);
+    private static Certificate[] loadCertificates(String certPath) throws IOException, CertificateException {
```

Comment:
The problem here that try-with-resources statement automatically closes input stream and close method throws `IOException` and then it goes to `addCA`.